File: bindings/node/src/smoke-abi.ts
import { abiMajorVersion, backendAvailable, version } from "./index.js";

const abiMajor = abiMajorVersion();
const abiVersion = version();
const kemAvailable = backendAvailable("ML-KEM-768");
const signatureAvailable = backendAvailable("ML-DSA-65");

console.log(
  JSON.stringify({
    abiMajor,
    abiVersion,
    kemAvailable,
    signatureAvailable,
  }),
);

File: bindings/python/smoke_abi.py
from __future__ import annotations

import json

import pqcb


print(
    json.dumps(
        {
            "abiMajor": pqcb.abi_major_version(),
            "abiVersion": pqcb.abi_version(),
            "kemAvailable": pqcb.backend_available("ML-KEM-768"),
            "signatureAvailable": pqcb.backend_available("ML-DSA-65"),
        },
        sort_keys=True,
    )
)
